### PR TITLE
Implemented refresh tokens

### LINKED ISSUE #160: Implement refresh tokens feature
## Description
As a user of the system,
I need refresh tokens,
so that I can refresh my access token to contribue using the backend services.

Read [this for more information on the need for refresh tokens.](https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/)
This came up because of this issue on the Android client issue tracker (https://github.com/systers/mentorship-android/issues/113)

Isabel's suggestion: This can be done with the already used [flask-jwt-extended](https://flask-jwt-extended.readthedocs.io/en/latest/) library.

### Update [Required]
- [ ] Implement refresh token feature

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
5 hours

### PR DESCRIPTION
**Pull request by iulianav**
### Description
Access tokens expire and require the user to login again in order to receive a new one. With the use of refresh tokens, the user can request from the auth server new access tokens without having to logout and login again. Moreover, the life of a refresh token is much longer, but for this reason they need to be stored very securely.

Fixes #160

### Type of Change:

- Code

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
I have not tested it properly yet.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by isabelcosta** [comment]
@iulianav thank you so much for sending this PR 🎉

I might have to study a little bit to understand how it works and review it properly, but from what I've seen in your PR your work seems to make a lot of sense 😄 

Regarding testing do you need help testing the API itself (1)? Or just writing tests for it (2)?
- (1) Do you know how to access Swagger UI? here's 2 wiki pages I wrote that kind of show how to use Swagger for [login](https://github.com/systers/mentorship-backend/wiki/Log-in-using-Swagger-UI) and [register new user](https://github.com/systers/mentorship-backend/wiki/Register-a-new-user-using-Swagger). When you run the app, you can access the Swagger UI through localhost:5000 (if I'm not mistaken). Do you need help with this?
- (2) I have no clear idea of how to do this. But you could look into this file [tests/users/test_api_login.py](https://github.com/systers/mentorship-backend/blob/develop/tests/users/test_api_login.py) and get some ideas?

Let me know if you have any question.

**Comment by iulianav** [comment]
@isabelcosta Thank you! I have already written some tests (2), but not sure about the location! At first, I wanted to add them to [test_api_login.py](https://github.com/systers/mentorship-backend/blob/develop/tests/users/test_api_login.py), but then I added them in the end to [test_api_authentication.py](https://github.com/systers/mentorship-backend/blob/develop/tests/users/test_api_authentication.py) because the class was called `TestProtectedAPI`? Maybe I should create a new test file, but I think it fits with the authentication.

I also played with (1) and it seemed fine. I will test it more tho!

**Comment by isabelcosta** [comment]
@iulianav Regarding written tests: I think it is better to create a separate file dedicated to your new API /refresh tests

There are some tests that may not be well separated, because when I did them I was still learning how to implement tests and make the project "modular".

**Comment by isabelcosta** [comment]
@iulianav I will edit te PR description to have `Fixes _#ISSUE_` instead of `Addresses _#ISSUE_`, because the first option closes the related issue when the PR is merged. Thats how the template is done ;)

**Comment by iulianav** [comment]
@isabelcosta great! I amended the commit following some API testing (1) and your suggestions. I believe everything will work as expected hopefully! :)

**Comment by isabelcosta** [comment]
@ramitsawhney27 , @m-murad can you take a look at this? I understand a bit of this feature, but I need more eyes on this just to be sure it does what is supposed to.

File: app/api/models/user.py
Comment (isabelcosta) [inline app/api/models/user.py:159]: I think this could remain still on this PR, until the android app is refactored to not use expiry, I'm not sure now, I think `expiry` is not used for anything on systers/mentorship-android if this is true than this can be removed :)

**Review by isabelcosta — DISMISSED** [review]
Great work @iulianav  with this PR!
I just need you to fix merge conflicts and then squash commits

You definitely followed the community guidelines here 🎉 
I just need at least one more review from maintainers here.

**Comment by iulianav** [comment]
@isabelcosta thank you alot!

File: app/api/models/user.py
Comment (iulianav) [reply app/api/models/user.py:159]: @isabelcosta i will leave it for now and add a TODO comment

File: app/api/models/user.py
Comment (iulianav) [inline app/api/models/user.py:161]: @isabelcosta i kept it.

Comment (isabelcosta) [reply app/api/models/user.py:161]: Cool ;) it can be removed later @iulianav

**Review by isabelcosta — APPROVED** [review]

**Pull request merged**
